stdlib/public/TensorFlow/TensorHandle.swift

```diff
@@ -16,22 +16,48 @@
 
 import CTensorFlow
 
+/// This protocol abstracts the underlying representation of a tensor. Any type
+/// that conforms to this protocol can be used as a `TensorHandle` in the
+/// `TensorFlow` library, as it much provide a way to convert the underlying tensor
+/// handle into a `ConcreteTensorHandle`, which wraps a `TFE_TensorHandle *`
+public protocol TFTensorHandle {
```

Comment:
Seems like TFTensorHandle and _AnyTensorHandle are redundant hierarchies? Did you just want to do that simplification later?

Reply:
You are right, they are redundant.  I wanted this change to be incremental. I actually started with changing _AnyTensorHandle to be a protocol, and adding a `handle: _AnyTensorHandle` property to `TensorHandle` variants. Unfortunately, that caused this change to propagate to more places. e.g., we rely on `TensorHandle` to be a base class of `_AnyTensorHandle` in the CompilerRuntime. @rxwei also raised the same issue. Let me take a look at this again.